### ISSUE TITLE
Pandas DataFrame to Javascript table wrapper

### ISSUE THREAD

**Issue by mikkokotila** [botlabio/autonomio#24]
In the background we have the data under control as a Pandas dataframe, and now we want to make it visible in the frontend using jquery datatable, or any other solution that does the job in a nice way. Should support up to around 100k rows of data at one go, maybe if it's too much it could be capped to lower for the table view. 

- support at least 20k rows of data and still work ok
- must be able to get the data from pandas backend
- must be able to sort by column
- must be able to export (we use it so can merge with predictions)

these seem to be standard features of javascript datatables. I have two code example ready, but I think there are many online.

**Comment by mikkokotila** [comment]
**sample of the json output from Pandas **

[json_from_pandas.json.zip](https://github.com/botlabio/autonomio/files/965212/json_from_pandas.json.zip)

**Comment by mikkokotila** [comment]
Here is the ready working DataTable that is in the same look and feel as our dashboard: 

[tables-datatables.html.zip](https://github.com/botlabio/autonomio/files/965928/tables-datatables.html.zip)

For CSV parsing, I think the best choice seems: 

http://papaparse.com/

It seems like a proper CSV parser, but javascript.

**Comment by sheepray** [comment]
```
support at least 20k rows of data and still work ok
```
That's a lot of data. Suggest move the computation to the backend and only show part of the data on UI.

**Comment by sheepray** [comment]
The same goes for
```
* must be able to get the data from pandas backend
* must be able to sort by column
```

**Comment by sheepray** [comment]
Since the Panda data frame is JSON already. Why do we need _CSV parsing_?

**Comment by sheepray** [comment]
[_pandas.DataFrame_](http://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.html) already provides huge useful API to manipulate data frame. It is unnecessary to move it to the browser. As mentioned above, we should move the heavy computation tasks to the Backend.

**Comment by mikkokotila** [comment]
Yes, we are not doing any processing in the frontend. Everything is done in the pandas dataframe, but pandas does not have option for browser display ;) What we want from the javascript is the sorting and exporting function, the data comes ready from pandas backend.
